server/conn.go

```diff
@@ -1519,10 +1528,9 @@ func (cc *clientConn) writeError(ctx context.Context, e error) error {
 // packets following it.
 // serverStatus, a flag bit represents server information
 // in the packet.
-func (cc *clientConn) writeEOF(serverStatus uint16) error {
+func (cc *clientConn) writeEOF(ctx context.Context, serverStatus uint16) error {
 	if cc.capability&mysql.ClientDeprecateEOF > 0 {
-		logutil.Logger(context.Background()).Error(
```

Comment:
Removing this error and calling `writeEOF()` would work, but I don't like this as this is not sending an EOF Packet, it is sending an OK Packet if ClientDeprecateEOF is set.

My intention was to eventually remove `writeEOF()` once support for older clients that don't set the flag would not be needed anymore.

Reply:
But looks like we will need `writeEOF()` in a long time. Also, it is OK packet with EOF indicator. So this is a natural change in my view.

Reply:
Yes that's true. While I have my preference; I'm fine with both options.

I would however suggest to fix the comment:
```patch
diff --git a/server/conn.go b/server/conn.go
index 8800d3eaf..cc87ebba1 100644
--- a/server/conn.go
+++ b/server/conn.go
@@ -1523,7 +1523,8 @@ func (cc *clientConn) writeError(ctx context.Context, e error) error {
        return cc.flush(ctx)
 }
 
-// writeEOF writes an EOF packet.
+// writeEOF writes an EOF packet or if ClientDeprecateEOF is set it
+// writes an OK packet with EOF indicator.
 // Note this function won't flush the stream because maybe there are more
 // packets following it.
 // serverStatus, a flag bit represents server information
```

Reply:
Done